multiqc/modules/ribotish/ribotish.py

```diff
@@ -0,0 +1,398 @@
+                    }
+
+        # Create configuration
+        pconfig = {
+            "id": "ribotish_frame_proportions",
+            "title": "RiboTish: Reading Frame Proportions by Read Length",
+            "ylab": "Proportion of Reads (%)",
+            "cpswitch_counts_label": "Counts",
+            "stacking": "normal",
+            "hide_zero_cats": False,
+            "ymax": 100,
+            "use_legend": True,
+            "sort_samples": False,
+        }
+
+        # Define category colors (Frame 0, 1, 2) matching viridis-like colors
+        cats = {
+            "Frame 0": {"name": "Frame 0", "color": "#440154"},
+            "Frame 1": {"name": "Frame 1", "color": "#31688e"},
+            "Frame 2": {"name": "Frame 2", "color": "#35b779"},
+        }
+
+        plot_html = bargraph.plot(plot_data, cats, pconfig)
+
+        self.add_section(
+            name="Reading Frame Proportions",
+            anchor="ribotish_frame_proportions",
+            description="Proportion of reads in each reading frame (Frame 0, 1, 2) for different read lengths (25-34nt). "
+            "High Frame 0 enrichment (>70%) indicates good quality Ribo-seq data.",
+            helptext="""
+            This plot shows the distribution of reads across the three reading frames for each read length.
+
+            * **Frame 0** (purple/dark): The correct reading frame - high values indicate good data quality
+            * **Frame 1** (blue): Off by 1 nucleotide
+            * **Frame 2** (green): Off by 2 nucleotides
+
+            For high-quality Ribo-seq data, Frame 0 should typically be >70% for read lengths 28-30nt.
+            Read lengths are shown as separate bars for each sample.
+            """,
+            plot=plot_html,
+        )
+
+    def add_read_length_heatmap(self):
+        """
+        Create a heatmap showing the percentage distribution of read lengths.
+        Rows are samples, columns are read lengths.
+        """
+        # Collect all read lengths
+        all_lengths = set()
+        for sample_data in self.frame_proportions.values():
+            all_lengths.update(sample_data.keys())
+        all_lengths = sorted(all_lengths)
+
+        # Prepare data for heatmap
+        samples = sorted(self.frame_proportions.keys())
+        heatmap_data = []
+
+        for sample_name in samples:
+            sample_total = sum(props["total"] for props in self.frame_proportions[sample_name].values())
+            row = []
+            for length in all_lengths:
+                if length in self.frame_proportions[sample_name]:
+                    count = self.frame_proportions[sample_name][length]["total"]
+                    percentage = (count / sample_total * 100) if sample_total > 0 else 0
+                    row.append(percentage)
+                else:
+                    row.append(0)
+            heatmap_data.append(row)
+
+        # Create heatmap configuration
+        pconfig = {
```

Comment:
Would benefit from a better colour scale here. A single colour gradient to indicate intensity / count.